src/main/java/no/unit/nva/doi/FetchDoiMetadata.java

```diff
@@ -0,0 +1,106 @@
+package no.unit.nva.doi;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.MalformedParametersException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Handler for requests to Lambda function.
+ */
+public class FetchDoiMetadata implements RequestHandler<String, Object> {
+
+    private static final Logger logger = LoggerFactory.getLogger(FetchDoiMetadata.class);
+
+    /** Defines how long we wait for datacite to answer */
+    private transient int external_service_timeout = 30_000;
+    /** datacite's URL to access json-formatted metadata  */
+    private final static transient String DATACITE_URL = "https://data.datacite.org/application/vnd.citationstyles.csl+json";
+
+    @Override
+    public Object handleRequest(String url, Context context) {
+        Map<String, String> headers = new ConcurrentHashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("X-Custom-Header", "application/json");
+        String json;
+        int statusCode;
+        try {
+            final String uri = getValidURI(url, 1024);
+            json = getDoiMetadataInJson(uri);
+            statusCode= 200;
+        } catch (URISyntaxException | MalformedURLException | UnsupportedEncodingException e) {
+            logger.warn(e.getMessage(), e);
+            statusCode = 400;
+            json = "{\"error\": \"" + e.getMessage() + "\"}";
```

Comment:
Since we're doing this multiple times and since inline generation of JSON as a string is a bit flaky, create a JSON object in a method